xapian-applications/omega/handler_libextractor.cc

```diff
@@ -0,0 +1,136 @@
+/** @file handler_libextractor.cc
+ * @brief Extract metadata using libextractor.
+ */
+/* Copyright (C) 2020 Parth Kapadia
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "handler.h"
+
+#include <extractor.h>
+#include <cstring>
+
+using namespace std;
+
+struct metadata {
+    string& title;
+    string& author;
+    string& keywords;
+    string& pages;
+
+    metadata(string& t, string& a, string& k, string& p)
+	    : title(t), author(a), keywords(k), pages(p) {}
+};
+
+/*
+* cls: passed as last parameter from EXTRACTOR_extract
+* plugin_name: name of the plugin
+* type: mime-type of file according to libextractor
+* format: format information about data
+* data_mime_type: mimetype of data according to libextractor
+* data: actual meta-data found
+* data_len: number of bytes in data
+*/
```

Comment:
For Xapian code, we'd usually indent the `*` and `*/` lines so the `*`s all line up vertically with the one in `/*`.

It's also better to format function parameter docs as a doxygen documentation comment - that way someone can run doxygen over the code and get collated docs for internal APIs.  Going through and fixing up existing docs that lack such formatting is tedious work which is hard to automate, so it's the sort of thing which doesn't ever get done, so doing this when the comments are added is better.

What that actually means is starting the comment block with `/**` followed by a brief description of the purpose of the function, and then formatting the parameter list entries like: `@param data_len  number of bytes in data`

Reply:
I have made these changes and updated the comment.